### DISCUSSION TITLE
Set Sentry.Protocol.Device and Sentry.Protocol.User

### DISCUSSION THREAD

**Discussion by dhpanteon** [getsentry/sentry-dotnet#896] [Q&A] [answered]
Please mark the type framework used:

- [x] Xamarin

Please mark the type of the runtime used:
- [x] Mono

Not able to set Sentry.Protocol.Device. 
How can i set Device and User on sentry?

**Comment by bruno-garcia** [comment]
@dhpanteon What do you mean you're not able to set.

Could you please try:

```csharp

SentrySdk.WithScope(s => {

s.Contexts.Device.Name = "name";
SentrySdk.CaptureMessage("test");
});
```

Does it work?
Or you're setting the device to the scope at a different point in the app like during startup? This could be related to the #628

Also, are you using Xamarin.Forms? @lucas-zimerman created an SDK for it, there's a preview version out already: https://github.com/getsentry/sentry-dotnet-xamarin

**Comment by lucas-zimerman** [comment, marked as answer]
I usually set the user the following way on Xamarin:

```csharp
            SentrySdk.ConfigureScope(s =>
            {
                s.User.Id = "userid...";
                s.User.Username = "username...";
                s.User.Email = "email...";
            });
```

Also on SentryOptions, for registering the user IP I set the SendDefaultPii to true.
